R/teal_data-extract.R

```diff
@@ -0,0 +1,33 @@
+#'
+#' @section Subsetting:
+#' `x[names]` subsets objects in `teal_data` environment and limit the code to the necessary needed to build limited
+#' objects.
+#'
+#' @param names (`character`) names of objects included in `teal_subset` to subset
+#' @param x (`teal_data`)
+#'
+#' @examples
+#'
+#' # Subsetting
+#' data <- teal_data()
+#' data <- eval_code(data, "a <- 1;b<-2")
+#' data["a"]
+#' data[c("a", "b")]
+#'
+#' join_keys(data) <- join_keys(join_key("a", "b", "x"))
+#' join_keys(data["a"]) # should show empty keys
+#' join_keys(data["b"])
+#' join_keys(data)["a"] # should show empty keys
+#' join_keys(data)["b"]
+#'
+#' @rdname teal_data
+#'
+#' @export
+`[.teal_data` <- function(x, names) {
+  x <- NextMethod("`[`", x, check_names = FALSE) # takes 'names' from function's environment
+  if (inherits(x, "qenv")) {
+    return(teal_data())
+  } # all 'names' not in object
```

Comment:
I always inherits from qenv
```suggestion
```

Reply:
Thanks! I needed an exit condition in case empty `qenv` is returned in `teal.code:::[.qenv`
I changed it to this https://github.com/insightsengineering/teal.data/pull/346/commits
Now teal works like a charm